lib/ae_mdw/db/origin.ex

```diff
@@ -96,6 +88,32 @@ defmodule AeMdw.Db.Origin do
     contract_id
   end
 
+  @spec count_contracts() :: non_neg_integer()
+  def count_contracts do
+    ct_list1 = do_list_by_tx_type([], {:contract_create_tx, <<>>, -1})
+    ct_list2 = do_list_by_tx_type([], {:contract_call_tx, <<>>, -1})
+
+    length(ct_list1) + length(ct_list2)
+  end
+
+  #
+  # Private functions
+  #
+  defp field_txi(tx_type, pos, pk) do
+    case next(Model.Field, {tx_type, pos, pk, -1}) do
+      :"$end_of_table" -> :error
+      {^tx_type, ^pos, ^pk, txi} -> {:ok, txi}
+      {_tx_type, _pos, _pk, _txi} -> :error
+    end
+  end
+
+  defp do_list_by_tx_type(origin_pks, {type, _pk, _txi} = key) do
+    case next(Model.Origin, key) do
+      {^type, pubkey, _txi} = next_key -> do_list_by_tx_type([pubkey | origin_pks], next_key)
+      _not_the_type -> origin_pks
+    end
```

Comment:
Same here, we can use the `Collection.stream/4` to iterate over a slice of consecutive keys from a table